core/src/main/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingService.scala

```diff
@@ -288,4 +304,43 @@ class SpendReportingService(
     }
   }
 
+  def getSpendForBillingProject(
+    project: RawlsBillingProjectName,
+    start: DateTime,
+    end: DateTime,
+    aggregations: Set[SpendReportingAggregationKeyWithSub]
+  ): Future[SpendReportingResults] =
+    for {
+      billingProject <- billingRepository.getBillingProject(project)
+
+      report <- getReportData(billingProject.get, project, start, end, aggregations)
+    } yield report
+
+  private def getReportData(billingProject: RawlsBillingProject,
+                            project: RawlsBillingProjectName,
+                            start: DateTime,
+                            end: DateTime,
+                            aggregations: Set[SpendReportingAggregationKeyWithSub]
+  ): Future[SpendReportingResults] =
+    if (billingProject.billingProfileId.isEmpty) {
+      getSpendForGCPBillingProject(project, start, end, aggregations)
+    } else {
+      getSpendForAzureBillingProject(billingProject.billingProfileId.get, start, end)
+    }
+
+  private def getSpendForAzureBillingProject(
+    billingProfileId: String,
+    start: DateTime,
+    end: DateTime
+  ): Future[SpendReportingResults] =
+    try {
+      val spendReport =
+        bpmDao.getAzureSpendReport(UUID.fromString(billingProfileId), start.toDate, end.toDate, ctx)
+
+      Future.successful(SpendReportingResultsConvertor(spendReport))
+    } catch {
+      case ex: BpmAzureSpendReportBadRequest =>
+        throw RawlsExceptionWithErrorReport(StatusCodes.BadRequest, ex.getMessage)
```

Comment:
@blakery Relating to our conversation at the end of standup… would it be better for this to return Future.failed(RawlsExceptionWithErrorReport..)? I'm unclear now about when it is important to return a failed future vs. just throwing the exception.